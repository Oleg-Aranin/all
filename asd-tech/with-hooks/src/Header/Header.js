import React, {useContext} from 'react'
import classes from './Header.module.css'
import {NavLink} from 'react-router-dom'
import {AppContext} from '../context/appContext'

export default () => {
const {state, signOut} = useContext(AppContext)
  return (
   <div className={classes.Header}>
      <NavLink
         to='/timeline'
         exact
         onClick={signOut} >

      { state.token ? 'Sign out' : 'Sign in' }

      </NavLink>
   </div>
)}
