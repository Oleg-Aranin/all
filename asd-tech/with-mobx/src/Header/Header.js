import React from 'react'
import classes from './Header.module.css'
import {NavLink} from 'react-router-dom'
import Store from '../store/store'
import {observer} from 'mobx-react'


const Header = observer(() => {
  let sign = Store.state.token
  return (
   <div className={classes.Header}>
      <NavLink
         to='/timeline'
         exact
         onClick={Store.signOut}
      >
      { sign ? 'Sign out' : 'Sign in' }
      </NavLink>
   </div>
)})

export default Header
