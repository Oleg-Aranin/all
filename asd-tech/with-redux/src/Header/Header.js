import React from 'react'
import classes from './Header.module.css'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'
import {signOut} from '../store/actions/actionHeader'


function Header(props) {

let sign = props.token

  return (
   <div className={classes.Header}>
      <NavLink
         to='/timeline'
         exact
         onClick={props.signOut}
          >
      { sign ? 'Sign out' : 'Sign in' }
      </NavLink>
   </div>
)}

function mapStateToProps(state) {
  return {
    token: state.app.token
  }
}

function mapDispatchToProps(dispatch) {
 return {
   signOut: () => dispatch(signOut())
 }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)
