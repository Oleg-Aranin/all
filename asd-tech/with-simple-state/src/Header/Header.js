import React from 'react'
import classes from './Header.module.css'
import {NavLink} from 'react-router-dom'

export default props => {

let sign = props.state

  return (
   <div className={classes.Header}>
      <NavLink
         to='/timeline'
         exact
         onClick={props.signOut}
          >

      { sign ? 'Sign out' : 'Sign in' }

      </NavLink>

      
   </div>
)}
