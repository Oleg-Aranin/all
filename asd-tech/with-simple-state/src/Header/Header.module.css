.Header *{
  margin: 10px;
}
