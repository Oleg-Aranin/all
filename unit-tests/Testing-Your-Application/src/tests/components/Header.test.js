import React from "react";
import {shallow, mount} from 'enzyme';
import Header from '../../components/Header'
import {Router, MemoryRouter} from "react-router-dom";

it('should Match Snapshot', () => {
    const wrapper = shallow(
        <MemoryRouter>
            <Header/>
        </MemoryRouter>
    )

    expect(wrapper.html()).toMatchSnapshot()
})

it('should render Header correctly', () => {
    const wrapper = shallow(<Header/>)

    expect(wrapper.find('h1').length).toBe(1)
    expect(wrapper.find('h1').text()).toBe('Expensify')
})
