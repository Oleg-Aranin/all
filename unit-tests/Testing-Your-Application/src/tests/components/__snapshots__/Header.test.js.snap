// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Match Snapshot 1`] = `"<header><h1>Expensify</h1><a aria-current=\\"page\\" class=\\"is-active\\" href=\\"/\\">Dashboard</a><a href=\\"/create\\">Create Expense</a><a href=\\"/help\\">Help</a></header>"`;

exports[`should render Header correctly 1`] = `"<header><h1>Expensify</h1><a aria-current=\\"page\\" class=\\"is-active\\" href=\\"/\\">Dashboard</a><a href=\\"/create\\">Create Expense</a><a href=\\"/help\\">Help</a></header>"`;
